pcapgo/ngread.go

```diff
@@ -0,0 +1,599 @@
+			return
+		}
+		//consume padding
+		padding := length % 4
+		if padding > 0 {
+			padding = 4 - padding
+			if _, err = r.r.Discard(int(padding)); err != nil {
+				return
+			}
+		}
+		r.currentBlock.length -= uint32(length + padding)
+	}
+	return
+}
+
+// readSectionHeader parses the full section header and implements section skipping in case of version mismatch
+// if needed, the first interface is read
+func (r *NgReader) readSectionHeader() (err error) {
+	if r.options.SectionEndCallback != nil && r.activeSection {
+		interfaces := make([]NgInterface, len(r.ifaces))
+		for i := range r.ifaces {
+			interfaces[i] = r.ifaces[i]
+		}
+		r.options.SectionEndCallback(interfaces, r.sectionInfo)
+	}
+	// clear the interfaces
+	r.ifaces = r.ifaces[:0]
+	r.activeSection = false
+
+RESTART:
+	// read major, minor, section length
+	if err = r.readBytes(r.buf[:12]); err != nil {
+		return
+	}
+	r.currentBlock.length -= 12
+
+	vMajor := r.getUint16(r.buf[0:2])
+	vMinor := r.getUint16(r.buf[2:4])
+	if vMajor != ngVersionMajor || vMinor != ngVersionMinor {
+		if !r.options.SkipUnknownVersion {
+			// Well the standard actually says to skip unknown version section headers,
+			// but this would mean user would be kept in the dark about whats going on...
+			return ErrNgVersionMismatch
+		}
+		if _, err = r.r.Discard(int(r.currentBlock.length)); err != nil {
+			return
+		}
+		if err = r.skipSection(); err != nil {
+			return
+		}
+		goto RESTART
+	}
+
+	var section NgSectionInfo
+
+	for {
+		if err = r.readOption(); err != nil {
+			return
+		}
+		switch r.currentOption.code {
+		case ngOptionCodeEndOfOptions:
+			goto DONE
+		case ngOptionCodeComment:
+			section.Comment = string(r.currentOption.value)
+		case ngOptionCodeHardware:
+			section.Hardware = string(r.currentOption.value)
+		case ngOptionCodeOS:
+			section.OS = string(r.currentOption.value)
+		case ngOptionCodeUserApplication:
+			section.Application = string(r.currentOption.value)
+		}
+	}
+DONE:
+	if _, err = r.r.Discard(int(r.currentBlock.length)); err != nil {
+		return
+	}
+	r.activeSection = true
+	r.sectionInfo = section
+
+	if !r.options.WantMixedLinkType {
+		// If we don't want mixed link type, we need the first interface to fill Reader.LinkType()
+		// This handles most of the pcapngs out there, since they start with an IDB
+		if err = r.firstInterface(); err != nil {
+			return
+		}
+	}
+
+	return
+}
+
+// skipSection skips blocks until the next section
+func (r *NgReader) skipSection() (err error) {
+	for {
+		if err = r.readBlock(); err != nil {
+			return err
+		}
+		if r.currentBlock.typ == ngBlockTypeSectionHeader {
+			return
+		}
+		if _, err = r.r.Discard(int(r.currentBlock.length)); err != nil {
+			return err
+		}
+	}
+}
+
+// SkipSection skips the contents of the rest of the current section and reads the next section header.
+func (r *NgReader) SkipSection() (err error) {
+	if err = r.skipSection(); err != nil {
+		return
+	}
+	err = r.readSectionHeader()
```

Comment:
just `return r.readSectionHeader()` ?

Reply:
Fixed by the named -> unnamed return value patch